apps/api/src/app/integrations/usecases/get-active-integration/get-active-integration.usecase.ts

```diff
@@ -1,20 +1,108 @@
 import { Injectable } from '@nestjs/common';
-import { IntegrationEntity } from '@novu/dal';
-import { GetDecryptedIntegrations, GetDecryptedIntegrationsCommand } from '@novu/application-generic';
 
-import { GetIntegrationsCommand } from '../get-integrations/get-integrations.command';
+import {
+  GetDecryptedIntegrations,
+  GetDecryptedIntegrationsCommand,
+  SelectIntegration,
+  SelectIntegrationCommand,
+} from '@novu/application-generic';
+import { ChannelTypeEnum } from '@novu/shared';
+import { IntegrationEntity, IntegrationRepository, EnvironmentRepository } from '@novu/dal';
+
+import { GetActiveIntegrationsCommand } from './get-active-integration.command';
+import { GetActiveIntegrationResponseDto } from '../../dtos/get-active-integration-response.dto';
 
 @Injectable()
 export class GetActiveIntegrations {
-  constructor(private getDecryptedIntegrationsUsecase: GetDecryptedIntegrations) {}
+  constructor(
+    private integrationRepository: IntegrationRepository,
+    private selectIntegration: SelectIntegration,
+    private environmentRepository: EnvironmentRepository,
+    private getDecryptedIntegrationsUsecase: GetDecryptedIntegrations
+  ) {}
 
-  async execute(command: GetIntegrationsCommand): Promise<IntegrationEntity[]> {
-    return await this.getDecryptedIntegrationsUsecase.execute(
+  async execute(command: GetActiveIntegrationsCommand): Promise<GetActiveIntegrationResponseDto[]> {
+    const activeIntegration = await this.getDecryptedIntegrationsUsecase.execute(
       GetDecryptedIntegrationsCommand.create({
         organizationId: command.organizationId,
-        active: true,
         userId: command.userId,
+        active: true,
       })
     );
+
+    if (!activeIntegration.length) {
+      return [];
+    }
+
+    const activeIntegrationChannelTypes = this.getDistinctChannelTypes(activeIntegration);
+    const selectedIntegrations = await this.getSelectedIntegrations(command, activeIntegrationChannelTypes);
+
+    return this.mapBySelectedIntegration(activeIntegration, selectedIntegrations);
+  }
+
+  private getDistinctChannelTypes(activeIntegration: IntegrationEntity[]): ChannelTypeEnum[] {
+    return activeIntegration.map((integration) => integration.channel).filter(this.distinct);
+  }
```

Comment:
```suggestion
 private getDistinctChannelTypes(activeIntegration: IntegrationEntity[]): ChannelTypeEnum[] {
    return Array.from(
      activeIntegration.reduce((acc, integration) => {
        acc.add(integration.channel);

        return acc;
      }, new Set<ChannelTypeEnum>())
    );
  }
```

Reply:
Is this one will be more readable in your opinion?
i wonder if we could create a util that we could reuse in order to get a distinct array, IMO this way one filter can do the job and should be readable.